base/linalg/bitarray.jl

```diff
@@ -123,7 +123,7 @@ end
 
 ## Structure query functions
 
-issymmetric(A::BitMatrix) = size(A, 1)==size(A, 2) && count(!iszero, A - transpose(A))==0
+issymmetric(A::BitMatrix) = size(A, 1)==size(A, 2) && count(!iszero, A - collect(A'))==0
```

Comment:
Just curious: is your rationale for `collect`ing here to ensure we still hit an optimized `-(::BitMatrix, ::BitMatrix)` method?  If that's the case it may be worth keeping note of this so we can revisit this PR for more `::Adjoint` method specializations in the future.

Reply:
Partly, yes :). During the relevant transformation pass, preserving behavior was my primary concern. And for some array types, the rewrite above was strictly necessary to preserve behavior. (For example, `S + Adjoint(S)` for sparse matrix `S` hits an `AbstractArray` `+` fallback yielding an `Array` rather than the former `S + adjoint(S)`'s `SparseMatrixCSC`. Hence the transformation `S + adjoint(S)` -> `S + collect(S')` rather than `S + S'`, though `S + S'` should work eventually.) And indeed, after the gross functionality is complete quite a bit of cleanup will be in order :). https://github.com/JuliaLang/julia/issues/25331 is my present breadcrumb for this particular issue.